test/Driver/EDMF/closures/mixing_length.jl

```diff
@@ -0,0 +1,119 @@
+#### Mixing length model kernels
+
+function mixing_length(
+    ss::SingleStack{FT, N},
+    m::MixingLengthModel,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+    δ::FT,
+    εt::FT,
+) where {FT, N}
+
+
+    # Q - do I need to define L as a vector ?
+    # need to code the functions: obukhov_length, ustar, ϕ_m, lamb_smooth_minimum
+
+    # Alias convention:
+    gm = state
+    en = state
+    up = state.edmf.updraft
+    gm_s = source
+    en_s = source
+    up_s = source.edmf.updraft
+    en_d = state.edmf.environment.diffusive
+
+    z = gm_a.z
+    # Parameters
+    cv_d::FT = cv_d(param_set)
+    cv_v::FT = cp_v(param_set)
+    cv_l::FT = cp_l(param_set)
+    cv_i::FT = cp_i(param_set)
+    T0::FT = T_0(param_set)
+    e_int_i0::FT = e_int_i0(param_set)
+    ϵ_v = 1. / molmass_ratio(param_set)
+    g = FT(grav(param_set))
+    R_d = FT(R_d(param_set))
+
+    ρinv = 1/gm.ρ
+    fill!(m.L, 0)
+    a_L = model.a_L(obukhov_length) # these functions are still missing 
+    b_L = model.b_L(obukhov_length) # these functions are still missing 
+
+    # precompute
+    en_area = 1-sum([up[i].ρa for i in 1:N])*ρinv
+    w_env = (gm.ρu[3]-sum([up[i].ρau[3] for i in 1:N]))*ρinv
+    en_e_int = (gm.ρe_int-up[i].ρae_int)/(gm.ρ*up_area)
+    en_q_tot = (gm.ρq_tot-up[i].ρaq_tot)/(gm.ρ*up_area)
+    TKE_Shear = en_d.∇u[1].^2 + en_d.∇u[2].^2 + en_d.∇u[3].^2
+    ∂e_int∂z = en_d.e_int
+    ∂q_tot∂z = en_d.q_tot
+
+
+    # Thermodynamic local variables for mixing length
+    ts = PhaseEquil(param_set ,en_e_int, gm.ρ, en_q_tot)
+    Π = exner(ts)
+    lv = latent_heat_vapor(ts)
+    cp_m_ = cp_m(ts)
+    tke = sqrt(max(en.ρatke, FT(0))*ρinv)
```

Comment:
Remove the max here. If this needs to be constrained, let’s use existing filters.